partner_communication/models/communication_job.py

```diff
@@ -269,19 +269,29 @@ def _get_default_vals(self, vals, default_vals=None):
         config = self.config_id.browse(vals['config_id'])
 
         # Determine user by default : take in config or employee
+        omr_config = config.omr_config_ids
         if not vals.get('user_id'):
-            vals['user_id'] = config.user_id.id or self.env.uid
-        user = self.env['res.users'].browse(vals['user_id'])
-        orm_config_of_right_lang = config.omr_config_ids \
-            .filtered(lambda c: c.lang_id.code == user.lang)
-        orm_config = orm_config_of_right_lang[0] if orm_config_of_right_lang \
-            else config.omr_config_ids
+            partner = self.env['res.partner'].browse(vals.get('partner_id'))
+            if partner:
+                lang_of_partner = self.env['res.lang'].search([
+                    ('code', 'like', partner.lang)
+                ])
+                omr_config = config.get_config_for_lang(lang_of_partner)
```

Comment:
```suggestion
                omr_config = config.get_config_for_lang(lang_of_partner)[0:]
```
In order to force take the first config in case there would be multiple configs with no language (that could cause a bug)